### ISSUE TITLE
Схемы

### ISSUE THREAD

**Issue by Fridom** [Fridom/TIMP#2]
Нарисовать раскладку куба с проиндексированными секторами.
Нарисовать блок-схему алгоритма распределения мин и чисел на кубе.
Нарисовать блок-схему события при клике левой кнопкой на блок с числом, или пустой блок, или мину(конец игры).
Нарисовать блок-схему при клике правой кнопкой мыши на блок.

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11749063/d91bd0bc-a055-11e5-8e37-476f458aaa0d.png)
Блок-схема события при клике любой кнопкой. Я перерисую и добавлю. 
А насчёт раскладки куба с проиндексированными секторами... Мне просто кубик надо нарисовать, я так полагаю?)

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11750017/482f375a-a05b-11e5-93fb-f9b486833585.png)
Левый клик

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11750078/a6855b40-a05b-11e5-9720-931eaca52664.png)
Левый клик

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11750119/e46dee9a-a05b-11e5-966c-c6fea98dabeb.png)
Третий раз. Левый клик. Делаем ставки, придётся ли мне переделывать ещё раз?)

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11750190/4704aa44-a05c-11e5-9281-675f817bbea7.png)
Левый клик

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11750201/51d46a40-a05c-11e5-8a73-5079b2e83fcb.png)
Правый клик

**Comment by SamsonicX** [comment]
"Нарисовать раскладку куба с проиндексированными секторами."
Вопрос: Сектора КУБА или сектора как клетки одной стороны?

**Comment by Fridom** [comment]
Да. Просто нарисовать разложенный куб, подписать, где какой face.
А вот как индексировать - сам решай, я не знаю как ты алгоритм построишь. Хотя помоему для каждой стороны отдельную индексацию потом обрабатывать будет проще (в плане того, что не придется присваивание номеров делать)

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11750624/8041adaa-a05e-11e5-9084-288f45bf567e.png)
Типо... так?

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11750889/2adad7ae-a060-11e5-9b06-90826a254676.png)
Надо сильнее распространять установку мин?

**Comment by Fridom** [comment]
нет не так.. Я имел ввиду, что у тебя есть раскладка как справа. И каждая сторона(Нормально подписаная) разбита на Конкретное число секторов, которые пронумерованы

**Comment by Fridom** [comment]
Насчет расстановки мин. Я думаю нам стоит ввести 2 случайных числа (0:6) и (0:кол-во ячеек на стороне) и рандомить положение каждой мины, если мина уже стоит, то рандомить еще раз, если на стороне больше чем n/5 мин, то рандомить еще раз

**Comment by SamsonicX** [comment]
думаю, лучше просто рандом и если упирается в мину, рандомить ещё раз

**Comment by Fridom** [comment]
тогда может получиться, что на одной стороне все мины. Лучше добавь условие

**Comment by SamsonicX** [comment]
Вероятность этого крайне мала. Может ввести счётчики на каждую сторону, как лимит количества мин на стороне?

**Comment by SamsonicX** [comment]
Как-то так
![](https://puu.sh/lRLDs/540c1c5ccd.png)

**Comment by Fridom** [comment]
норм

**Comment by Fridom** [comment]
теперь на основе этого подумай, как будут считаться числа на стыках

**Comment by SamsonicX** [comment]
![](https://puu.sh/lRLNe/d68673370c.png)
А чего тут думать. Сейчас буду писать алгоритм

**Comment by SamsonicX** [comment]
![image](https://cloud.githubusercontent.com/assets/6678241/11751567/8633fa1e-a064-11e5-9b1f-34c6b4255a2d.png)

**Comment by Fridom** [comment]
это ты к чему?

**Comment by SamsonicX** [comment]
Переделываю блок-схемы
